homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,257 @@
+"""Support for Watergate sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import UTC, datetime
+import logging
+
+from watergate_local_api.models import AutoShutOffReport
+
+from homeassistant.components.sensor import (
+    HomeAssistant,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfPressure,
+    UnitOfTemperature,
+    UnitOfTime,
+    UnitOfVolume,
+    UnitOfVolumeFlowRate,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WatergateConfigEntry
+from .coordinator import WatergateAgregatedRequests, WatergateDataCoordinator
+from .entity import WatergateEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+
+@dataclass(kw_only=True, frozen=True)
+class WatergateSensorEntityDescription(SensorEntityDescription):
+    """Description for Watergate sensor entities."""
+
+    value_fn: Callable[[WatergateAgregatedRequests], str | int | float | None]
+
+
+DESCRIPTIONS: list[WatergateSensorEntityDescription] = [
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.water_meter.duration
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_volume",
+        key="water_meter_volume",
+        native_unit_of_measurement=UnitOfVolume.LITERS,
+        device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.state.water_meter.duration
+        if data.state and data.state.water_meter
+        else None,
+        translation_key="water_meter_duration",
+        key="water_meter_duration",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.ip if data.networking else None,
+        translation_key="ip",
+        key="ip",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.gateway if data.networking else None,
+        translation_key="gateway",
+        key="gateway",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+    ),
+    WatergateSensorEntityDescription(
+        value_fn=lambda data: data.networking.subnet if data.networking else None,
+        translation_key="subnet",
```

Comment:
why would this information be interesting to have in HA? I can't think of something to automate with this.

Reply:
This is just diagnostic to see all the stuff that are related to the device. We are not showing this in the application - just yet but we will be. Without that you will not be able to see your device network configuration.